polyply/tests/test_nb_engine.py

```diff
@@ -0,0 +1,208 @@
+             except IndexError:
+                continue
+
+     nb_engine =  NonBondEngine.from_topology(topology.molecules,
+                                              topology,
+                                              box=np.array([10., 10., 10.]))
+
+     # now we check if the 8th is added correctly
+     nb_engine.remove_positions(mol_idx=2, node_key=0)
+     assert all(nb_engine.get_point(2, 0) == np.array([np.inf, np.inf, np.inf]))
+
+def test_update_positions_in_molecules(topology):
+     # we add 15 random positions to all molecules
+     positions = np.random.random(5*3).reshape(-1, 3)
+     for mol_idx, mol in enumerate(topology.molecules):
+         for node in mol.nodes:
+             print(mol_idx, node)
+             mol.nodes[node]["position"] = positions[mol_idx + node]
+
+     nb_engine =  NonBondEngine.from_topology(topology.molecules,
+                                              topology,
+                                              box=np.array([10., 10., 10.]))
+     # now we overwrite those positoins one
+     nb_engine.positions[:, :] = np.ones((5, 3), dtype=float)[:, :]
+     # then we update them in the molecules and check if they are actually updated
+     nb_engine.update_positions_in_molecules(topology.molecules)
+     for mol_idx, mol in enumerate(topology.molecules):
+         for node in mol.nodes:
+            assert all(mol.nodes[node]["position"] == np.array([1., 1., 1.]))
+
+@pytest.mark.parametrize('mol_idx_a, mol_idx_b, node_a, node_b, expected',
+                        ((0, 0, 0, 1, (0.53+0.67)/2.0),
+                         (0, 2, 0, 0, (0.43+0.53)/2.0),
+                         (0, 2, 1, 0, (0.43+0.67)/2.0)))
+def test_get_interaction(topology, mol_idx_a, mol_idx_b, node_a, node_b, expected):
+     # initiate the nb_engine
+     nb_engine =  NonBondEngine.from_topology(topology.molecules,
+                                              topology,
+                                              box=np.array([10., 10., 10.]))
+
+     # check get interactions works properly
+     value = nb_engine.get_interaction(mol_idx_a=mol_idx_a,
+                                       mol_idx_b=mol_idx_b,
+                                       node_a=node_a,
+                                       node_b=node_b)
+     assert value == (expected, 1.0)
+
+@pytest.mark.parametrize('dist, ref, expected',
+                        # LJ force is zero at minimum
+			((2**(1/6.)*0.35,
+                          np.array([0.0, 0.0, 2**1/6.*0.35]),
+                          np.array([0.0, 0.0, 0.0])),
+                        # LJ force is also zero at infinity
+                         (100000.0,
+                          np.array([0.0, 0.0, 100000.0]),
+                          np.array([0.0, 0.0, 0.0])),
+                        # pick one point inbetween
+                         (0.42,
+                          np.array([0.0, 0.0, 0.42]),
```

Comment:
Why is it always the last dimension that's dist?

Reply:
I guess I was thinking of them along the z-axis. But I can mix this up a little.